### ISSUE TITLE
Custom template packs in {% crispy %}

### ISSUE THREAD

**Issue by dmpayton** [django-crispy-forms/django-crispy-forms#142]
Howdy,

A web app I'm building requires the ability to dynamically pass in the template pack to the crispy tag (e.g., depending on the user-selected theme). However when I pass the template pack as a variable...

```
{% crispy form form.helper template_pack %}
```

...I get an exception...

```
TemplateSyntaxError: crispy tag's template_pack argument should be in quotes
```

...and when I hard-code as a string...

```
{% crispy form form.helper "template_pack" %}
```

...I get another exception...

```
TemplateSyntaxError: crispy tag's template_pack argument should be one of 'bootstrap' or 'uni_form'
```

Not being able to specify a custom template pack in the crispy tag feels broken to me. I dug into the code a bit, and it looks like custom template packs are only supported via the `CRISPY_TEMPLATE_PACK` setting, which means it's site-wide.

The root of the issue seems to be some ambiguity in the arguments of the crispy tag, namely that it supports both `{% crispy form helper %}` and `{% crispy form "template_pack" %}`.

The tag puts some effort into determining what the second argument actually is when there's only two arguments, and this is the reason why the available template packs are hardcoded to only the pre-built template packs.

I have a patch which removes this ambiguity and allows you to pass in a template variable or string as the template pack, allowing you to use any template pack. I feel like the resulting code is simpler and brings the tag more inline with the Zen of Python ("Explicit is better than implicit.", "In the face of ambiguity, refuse the temptation to guess.")

The downside is that this is not strictly backwards compatible. You would now be required to explicitly pass in the form helper (or None) when specifying a template pack:

```
{% crispy form form.helper template_pack %}
```

```
{% crispy form None "template_pack" %}
```

I did need to update one test (`test_template_pack_override`) to reflect this.

I'm going to hold off on issuing a pull-request because I'm mainly looking for feedback (also docs need to be updated).

**TL;DR** -- Patched {% crispy %} to allow custom template packs as a variable or string, resulting in less ambiguity and a more PEP-20 compliant tag, but is slightly backwards incompatible. Looking for feedback.

Code is here: https://github.com/dmpayton/django-crispy-forms/tree/crispy-tag-template-packs

Thanks!
Derek
